index.bs

```diff
@@ -199,23 +199,35 @@ Abstract Operations {#abstract-operations}
 
 Automation {#automation}
 ==========
-This section extends the [=automation=] section defined in the Generic Sensor API [[GENERIC-SENSOR]]
-to provide mocking information about the proximity level for the purposes of
-testing a user agent's implementation of {{ProximitySensor}} API.
 
-<h3 id="mock-proximity-sensor-type">Mock Sensor Type</h3>
+This section extends [[GENERIC-SENSOR#automation]] by providing [=Proximity Sensor=]-specific virtual sensor metadata.
 
-The {{ProximitySensor}} class has an associated [=mock sensor type=] which is
-<a for="MockSensorType" enum-value>"proximity"</a>, its [=mock sensor reading values=]
-dictionary is defined as follows:
+The [=per-type virtual sensor metadata=] [=map=] must have the following [=map/entry=]:
+: [=map/key=]
+:: "`proximity`"
+: [=map/value=]
+:: A [=virtual sensor metadata=] whose [=virtual sensor metadata/virtual sensor type=] is [=Proximity Sensor=] and [=reading parsing algorithm=] is the [=proximity reading parsing algorithm=].
 
-<pre class="idl">
-  dictionary ProximityReadingValues {
-    required double? distance;
-    required double? max;
-    required boolean? near;
-  };
-</pre>
+<h3 dfn>Proximity reading parsing algorithm</h3>
+<div algorithm="proximity reading parsing algorithm">
+    : input
+    :: |parameters|, a JSON {{Object}}
+    : output
+    :: A [=sensor reading=] or **undefined**
+
+    1.  Let |reading| be a new [=sensor reading=].
+    1.  Let |keys| be the [=/list=] « "`distance`", "`max`" ».
+    1.  [=list/For each=] |key| of |keys|
+        1.  Let |value| be the result of [=parse single-value number reading=] with |parameters| and |key|.
+            1.  If |value| is **undefined**.
+                1.  Return **undefined**.
+        1.  [=map/Set=] |reading|[|keys|] to |value|[|keys|].
+    1.  Let |near| be the result of invoking [=get a property=] from |parameters| with "near".
```

Comment:
```suggestion
    1.  Let |near| be the result of invoking [=get a property=] from |parameters| with "`near`".
```

Reply:
Fixed in https://github.com/w3c/proximity/pull/57/commits/0e039e64e6b1746dc5e34e2cf010be713f8a3e98
Could you double check the correctness of line 224.

Reply:
> Could you double check the correctness of line 224.

Good catch, I've added a suggested fix to the line.